### PR TITLE
Added onCapture function parameter, and customisable capture button

### PR DESCRIPTION
**Pull request by Shamsudeen12**
## Description
This PR adds a function parameter "onCapture" to allow the user to perform an action immediately after taking a photo. It also allows the user customise the capture button
*What your Pull Request change ?*

## Checklist

Before creating any Pull Request, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).

- [ x] 📕 I read the [Contributing page](https://github.com/Apparence-io/camera_awesome/blob/master/CONTRIBUTING.md).
- [x ] 🤝 I match the actual coding style.
- [x ] ✅ I ran ```flutter analyze``` without any issues.

## Breaking Change

- [ ] 🛠 My feature contain breaking change.

*If your feature break something, please detail it*

### CODE REVIEW

File: lib/src/orchestrator/states/photo_camera_state.dart [hunk 1]
```diff
@@ -63,7 +62,7 @@ class PhotoCameraState extends CameraState {
   ///
   /// You can listen to [cameraSetup.mediaCaptureStream] to get updates
   /// of the photo capture (capturing, success/failure)
-  Future<String> takePhoto() async {
+  Future<String> takePhoto(Function(String)? onCapture) async {
```
Comment (g-apparence) [inline lib/src/orchestrator/states/photo_camera_state.dart:65]: - Could you put this into named parameter 
- typedef a OnPhotoCallback
``` Future<String> takePhoto({OnPhotoCallback? onCapture}) async {```

**Review by g-apparence — COMMENTED** [review]
Thank you for your PR. 
I've added a few comments. 
Also can you run dart format on your pull request (this prevent github showing false changes and pubdev doens't like plugins/packages not formatted with the dart formatter).

**Comment by Shamsudeen12** [comment]
Alright, will do @g-apparence

**Comment by Shamsudeen12** [comment]
@g-apparence I have made the changes you requested

**Review by g-apparence — APPROVED** [review]

**Comment by Juanchote** [comment]
when are you guys going to merge this into master?

**Pull request merged**